pylmod/gradebook.py

```diff
@@ -22,17 +31,50 @@ def __init__(
     ):
         super(GradeBook, self).__init__(cert, urlbase)
         if gbuuid is not None:
-            self.gradebookid = self.get_gradebook_id(gbuuid)
+            self.gradebook_id = self.get_gradebook_id(gbuuid)
 
     def get_gradebook_id(self, gbuuid):
         """return gradebookid for a given gradebook uuid."""
-        gradebook_id = self.get('gradebook', params={'uuid': gbuuid})
-        if 'data' not in gradebook_id:
-            log.info(gradebook_id)
-            msg = "[PyLmod] error in get_gradebook_id - no data"
-            log.info(msg)
-            raise Exception(msg)
-        return gradebook_id['data']['gradebookId']
+        gradebook = self.get('gradebook', params={'uuid': gbuuid})
+        if 'data' not in gradebook:
+            failure_messsage = ('Error in get_gradebook_id '
+                                'for {0} - no data'.format(
+                                    gradebook
+                                ))
+            log.error(failure_messsage)
+            raise PyLmodUnexpectedData(failure_messsage)
+        return gradebook['data']['gradebookId']
+
+    def get_assignments(
+            self,
+            gradebook_id='',
+            simple=False,
+            max_points=True,
+            avg_stats=False,
+            grading_stats=False
```

Comment:
Question: You added these parameters to the function signature, which makes me wonder, "Why weren't they there to begin with?".